### PR TITLE
Adding new incoming call event for call automation. Related work item…

### PR DESCRIPTION
**Pull request by ramneettung-msft**
…: 3954169

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by danishgh — APPROVED** [review]

**Review by antonsamson-msft — APPROVED** [review]

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /// The add participant failed event.
+    /// </summary>
+    public class IncomingCall : CallAutomationEventBase
+    {
+        /// <summary> Initializes a new instance of <see cref="IncomingCall"/>. </summary>
+        internal IncomingCall()
+        {
+        }
+
+        /// <summary> Initializes a new instance of IncomingCallEvent. </summary>
```
Comment (antonsamson-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs:18]: Nit: `IncomingCall` instead?

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        /// <param name="internalEvent">Internal Representation of the AddParticipantFailedEvent. </param>
```
Comment (antonsamson-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs:19]: Nit: typo, should be IncomingCall?

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs (hunk 1 above)
Comment (ramneettung-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs:18]: Will change

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs (hunk 2 above)
Comment (ramneettung-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/IncomingCall.cs:19]: Will change

**Review by minwoolee-msft — APPROVED** [review]

**Review by antonsamson-msft — APPROVED** [review]

**Pull request merged**